packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx

```diff
@@ -0,0 +1,159 @@
+
+They work by taking in a DOM element. They render and attach the React Composite to that element. This means
+that you can use the UI Composites in any application that can render HTML elements to the browser DOM.
+
+**Note:** Something to keep in mind when using these loader functions for the UI Composites is that you are taking on
+not just the bundle for `@azure/communication-react` but also the bundle for `react` and `react-dom`. Best practice here
+is to lazy load these chunks so that you don't get caught waiting for the large bundle when you don't need your
+composite experience yet.
+
+This can be done like so:
+
+```ts
+...
+/**
+ * This is how you would load in the CallComposite with lazy loading. This goes in
+ * line in whatever function might initiate your call experience.
+ */
+*/
+import('@azure/communication-react/javascript-loaders').then(async (uiLib) => {
+  this.adapter = await uiLib
+    .loadCallComposite() // <- you can then use the loader functions here.
+...
+```
+
+## Setup needed before calling the functions
+
+All of the loaders are not included on the main API for the UI library. To better support our other customers using supported
+versions of React, we are providing these loaders in a submodule of `@azure/communication-react`. To use these loaders you will import
+from a different path than the main UI library.
+
+```ts
+import {
+  loadChatComposite,
+  loadCallComposite,
+  loadCallWithChatComposite,
+  loadOutboundCallComposite
+} from '@azure/communication-react/javascript-loaders';
+```
+
+Since these functions are imported from a submodule you might need to make some changes to your
+`tsconfig.json` to [handle the **submodule**](https://www.typescriptlang.org/tsconfig/#module). To do this you will need to set the following fields as follows
+
+```json
+"compilerOptions": {
+  "module": "NodeNext",
+  "moduleResolution": "NodeNext",
+}
+```
+
+Once you have set these you will be able to import them from the submodule. You do not need to state the
+submodule in your `package.json` as the submodule is included when you define you `@azure/communication-react`
+dependency.
+
+## Using the Loaders
+
+The following snippet is an example of how you might import the `CallComposite` in your [Angular](https://angular.dev/) application.
+
+```ts
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'app-calling',
+  standalone: true,
+  templateUrl: './calling.component.html',
+  styleUrls: ['./calling.component.css']
+})
+export class CallingComponent {
+  userId = '<ACS User Id>';
+  token = '<ACS Token>';
+  displayName = 'User';
+  groupId = '<GUID for call>';
+  adapter: any;
+
+  /**
+   * This is how you would load in the CallComposite with lazy loading. This
+   * will only load the CallComposite and the UI library when the function is called.
+   */
+  async loadCalling(): Promise<void> {
+    import('@azure/communication-react/javascript-loaders').then(async (uiLib) => {
```

Comment:
these all lines could be just all async/await style for linear logic

const uiLib = await import('@azure/communication-react/javascript-loaders');
this.adapter = await uiLib.loadCallComposiet(...)

this.adapter.on(...)

Reply:
yup totally, do we have a preference here on how we handle the Promise?